Tests/FluentAssertions.Equivalency.Specs/SelectionRulesSpecs.cs

```diff
@@ -602,6 +602,86 @@ public void When_both_field_and_properties_are_configured_for_inclusion_both_sho
             // Assert
             act.Should().Throw<XunitException>().Which.Message.Should().Contain("Field1").And.Contain("Property1");
         }
+
+        [Fact]
+        public void Taking_internal_fields_into_consideration()
+        {
+            // Arrage
+            var subject = new ClassWithAllAccessModifiersForMembers("Lorem", "protectedValue", "ipsum", "protectedInternalValue", "privateValue", "privateProtectedValue");
+            var expected = new ClassWithAllAccessModifiersForMembers("Lorem", "protectedValue", "ipsum", "protectedInternalValue", "privateValue", "privateProtectedValue");
+
+            // Act / Assert
+            subject.Should().BeEquivalentTo(expected, opts => opts.IncludingInternalFields());
+        }
+
+        [Fact]
+        public void Taking_internal_fields_into_consideration_throws()
+        {
+            // Arrage
+            var subject = new ClassWithAllAccessModifiersForMembers("Lorem", "protectedValue", "ipsum", "protectedInternalValue", "privateValue", "privateProtectedValue");
+            var expected = new ClassWithAllAccessModifiersForMembers("Lorem", "protectedValue", "dolor", "protectedInternalValue", "privateValue", "privateProtectedValue");
+
+            // Act
+            Action act = () => subject.Should().BeEquivalentTo(expected, opts => opts.IncludingInternalFields());
+
+            // Assert
+            act.Should().Throw<XunitException>();
+        }
+
+        [Fact]
+        public void Taking_nested_objects_into_account()
+        {
+            // Arrange
+            var subject = new
+            {
+                A = new
+                {
+                    B = "C"
+                },
+                B = "D"
+            };
+
+            var expectation = new
+            {
+                A = new
+                {
+                    B = "C"
+                },
+                B = "D"
+            };
+
+            // Act / Assert
+            subject.Should().BeEquivalentTo(expectation, opts => opts.IncludingNestedObjects());
+        }
+
+        [Fact]
+        public void Taking_nested_objects_into_account_throws()
```

Comment:
🤔 It's not that it throws because nested objects are included, but because the nested objects differ. By the way, this the default.

Reply:
Hmm... true

If this is the default.. how can I construct a test case then?

What comes to my mind first: 
Run two assertions one with excluded and one with included and check if they are the same. 🤔

Reply:
Yeah, something like that. Although I agree it looks a bit weird. Maybe it's enough to change the title.